src/dependencies.rs

```diff
@@ -78,6 +78,26 @@ fn cfgs(config: &Config) -> Result<Vec<Cfg>, Errored> {
     Ok(cfgs)
 }
 
+pub(crate) fn has_build_std(info: &DependencyBuilder) -> bool {
```

Comment:
I think it would be simpler to have a `build_std` bool field on `DependencyBuilder` and manually inject the `-Z` flags instead of expecting the user to supply them and have us detect them

Reply:
This couldn't be a `bool` because sometimes its relevant to pass arguments like [`-Zbuild-std=std,panic_abort`](https://doc.rust-lang.org/1.78.0/cargo/reference/unstable.html#build-std).

But I could instead make it an `Option<String>`?

I think another motivation to not add a configuration is that when https://github.com/rust-lang/wg-cargo-std-aware/issues/20 is addressed, we wouldn't need a configuration either. 

Let me know how you want me to proceed.

Reply:
> But I could instead make it an `Option<String>`?

yea that seems good



> I think another motivation to not add a configuration is that when [rust-lang/wg-cargo-std-aware#20](https://github.com/rust-lang/wg-cargo-std-aware/issues/20) is addressed, we wouldn't need a configuration either.

I'm doing regular major bumps anyway, seems fine to do another once that lands (or just mark it as deprecated then)

Reply:
Done.

After trying it out though I have to say its a bit weird, because now **if** the user supplies the `-Zbuild-std` argument without enabling the `DependencyBuilder::build_std` option, it won't work.